### DISCUSSION TITLE
How do I get nginx to run the mastodon configuration?

### DISCUSSION THREAD

**Discussion by interN3rd** [mastodon/mastodon#21041] [Q&A] [answered]
I opened an issue here: https://github.com/mastodon/mastodon/issues/21032 But I think, opening an issue was a bit over the top. So, next try on this place. I run a Ubuntu 20.04 VPS and when running the command `systemctl status nginx.service` it responds as the following:

```
 nginx.service - nginx - high performance web server
     Loaded: loaded (/lib/systemd/system/nginx.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2022-11-18 20:39:50 CET; 12s ago
       Docs: https://nginx.org/en/docs/
    Process: 23733 ExecStart=/usr/sbin/nginx -c /etc/nginx/nginx.conf (code=exited, status=0/SUCCESS)
   Main PID: 23734 (nginx)
      Tasks: 7 (limit: 105)
     Memory: 5.2M
     CGroup: /system.slice/nginx.service
             ├─23734 nginx: master process /usr/sbin/nginx -c /etc/nginx/nginx.conf
             ├─23735 nginx: worker process
             ├─23736 nginx: worker process
             ├─23737 nginx: worker process
             ├─23738 nginx: worker process
             ├─23739 nginx: worker process
             └─23740 nginx: worker process
```
I assume the cause of my problems is in ` Process: 23733 ExecStart=/usr/sbin/nginx -c /etc/nginx/nginx.conf (code=exited, status=0/SUCCESS)`. This is the default config that seems to be running and not the configuration file that is provided during the mastodon installation process. How do I change it? Why is it happening? And why is none of the nginx configuration mentioned in https://docs.joinmastodon.org/admin/install/?

The overall problem I have is: I can't run Mastodon at all. When visiting mydomain.com I get a nginx default page and when visiting the https version of mydomain.com I don't get anything, the browser can't even establish a connection

**Comment by scytaleqc** [comment]
Have you disabled nginx's default site in `/etc/nginx/sites-available/`  ?

Then do `nginx -t`

**Reply by interN3rd** [reply]
If by "disabled" you mean deleted, then yes :) There is only mastodon. And when running `nginx -t` it also says that everything was fine. Here's the output:

```
nginx: the configuration file /etc/nginx/nginx.conf syntax is ok
nginx: configuration file /etc/nginx/nginx.conf test is successful
```

So it also refers to the default configuration and not to the mastodon configuration. I can't even run `$EDITOR /etc/systemd/system/mastodon-*.service`, because it says 'Permission denied'

**Comment by interN3rd** [comment, marked as answer]
It turns out, that it is necessary to change the `/etc/nginx/nginx.conf` to NOT `include /etc/nginx/conf.d/*.conf;`, but to `include /etc/nginx/sites-enabled/*;` in order to make the sites-enabled approach work, which is suggested in the official documentation. Thanks to @afontenot who pointed that out in https://github.com/mastodon/mastodon/issues/21032
